# Assignment 2:Think Like a Data Analyst
**Module:** Fundamentals of AI and Data Analytics  
**Professor:** Lucas Silva  
**Dataset:** Movies Dataset

**Initial Description**: The movies dataset is a dataset which is organized in rows and columns and it is downloaded from Kaggle. The dataset has 4000 rows and 15 columns and each row has one movie that has been released from the years of 1981 to 2001. The dataset includes a mix of variable types ranging 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

In [ ]:
df = pd.read_csv('movies_updated.csv')
df.head()

,name,rating,genre,year,released,score,votes,director,writer,star,country,budget,gross,company,"runtime,,"
0,The Shining,R,Drama,1980,"June 13, 1980 (United States)",8.4,927000,Stanley Kubrick,Stephen King,Jack Nicholson,United Kingdom,19000000,46998772.0,Warner Bros.,"146.0,"
1,The Blue Lagoon,R,Adventure,1980,"July 2, 1980 (United States)",5.8,65000,Randal Kleiser,Henry De Vere Stacpoole,Brooke Shields,United States,4500000,58853106.0,Columbia Pictures,"104.0,"
2,Star Wars: Episode V - The Empire Strikes Back,PG,Action,1980,"June 20, 1980 (United States)",8.7,1200000,Irvin Kershner,Leigh Brackett,Mark Hamill,United States,18000000,538375067.0,Lucasfilm,"124.0,"
3,Airplane!,PG,Comedy,1980,"July 2, 1980 (United States)",7.7,221000,Jim Abrahams,Jim Abrahams,Robert Hays,United States,3500000,83453539.0,Paramount Pictures,"88.0,"
4,Caddyshack,R,Comedy,1980,"July 25, 1980 (United States)",7.3,108000,Harold Ramis,Brian Doyle-Murray,Chevy Chase,United States,6000000,39846344.0,Orion Pictures,"98.0,"


In [3]:
df.head()

,name,rating,genre,year,released,score,votes,director,writer,star,country,budget,gross,company,"runtime,,"
0,The Shining,R,Drama,1980,"June 13, 1980 (United States)",8.4,927000,Stanley Kubrick,Stephen King,Jack Nicholson,United Kingdom,19000000,46998772.0,Warner Bros.,"146.0,"
1,The Blue Lagoon,R,Adventure,1980,"July 2, 1980 (United States)",5.8,65000,Randal Kleiser,Henry De Vere Stacpoole,Brooke Shields,United States,4500000,58853106.0,Columbia Pictures,"104.0,"
2,Star Wars: Episode V - The Empire Strikes Back,PG,Action,1980,"June 20, 1980 (United States)",8.7,1200000,Irvin Kershner,Leigh Brackett,Mark Hamill,United States,18000000,538375067.0,Lucasfilm,"124.0,"
3,Airplane!,PG,Comedy,1980,"July 2, 1980 (United States)",7.7,221000,Jim Abrahams,Jim Abrahams,Robert Hays,United States,3500000,83453539.0,Paramount Pictures,"88.0,"
4,Caddyshack,R,Comedy,1980,"July 25, 1980 (United States)",7.3,108000,Harold Ramis,Brian Doyle-Murray,Chevy Chase,United States,6000000,39846344.0,Orion Pictures,"98.0,"


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 4000 entries, 0 to 3999
Data columns (total 15 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   name       4000 non-null   str    
 1   rating     3960 non-null   str    
 2   genre      4000 non-null   str    
 3   year       4000 non-null   int64  
 4   released   4000 non-null   str    
 5   score      4000 non-null   float64
 6   votes      4000 non-null   int64  
 7   director   4000 non-null   str    
 8   writer     3999 non-null   str    
 9   star       3999 non-null   str    
 10  country    4000 non-null   str    
 11  budget     4000 non-null   int64  
 12  gross      3831 non-null   float64
 13  company    3990 non-null   str    
 14  runtime,,  4000 non-null   str    
dtypes: float64(2), int64(3), str(10)
memory usage: 468.9 KB


In [5]:
df.shape

(4000, 15)

**Data Quality Issues**

Missing Values

In [6]:
df.isna().sum()

name           0
rating        40
genre          0
year           0
released       0
score          0
votes          0
director       0
writer         1
star           1
country        0
budget         0
gross        169
company       10
runtime,,      0
dtype: int64

Missing values were found in 5 columns which are rating, writer, star, gross and company.

Duplicates

In [8]:
df.duplicated().sum()

np.int64(0)

this means that no duplicate values were found

Inconsistent categories

In [9]:
df['genre'].value_counts()

genre
Comedy       1355
Drama         807
Action        775
Crime         302
Adventure     231
Biography     175
Horror        158
Animation     113
Fantasy        33
Mystery        13
Thriller       11
Family          8
Romance         8
Sci-Fi          6
Western         3
Music           1
History         1
Name: count, dtype: int64

All genre categories are clean 

In [10]:
df['rating'].value_counts()

rating
R            2043
PG-13         848
PG            815
G             106
Not Rated      92
Unrated        32
NC-17          14
TV-MA           4
X               3
TV-PG           2
Approved        1
Name: count, dtype: int64

There are some inconsistent categories here. The main ratings are R,PG13, PG and G. but the other values from 'not rated' to 'approved' are inconsistent and dont align with the main ratings.

In [11]:
df['country'].value_counts()

country
United States                     2972
United Kingdom                     396
France                             150
Canada                             100
Australia                           53
Italy                               45
Japan                               45
Germany                             43
Hong Kong                           26
Spain                               19
Ireland                             18
West Germany                        12
Mexico                              12
Denmark                             11
Sweden                              10
New Zealand                         10
Switzerland                          8
Netherlands                          7
Iran                                 7
Taiwan                               6
India                                6
Yugoslavia                           5
South Africa                         4
China                                4
South Korea                          3
Hungary          

Country categories are consistent with no typos or errors.

Outliers

In [12]:
Q1 = df['score'].quantile(0.25)
Q3 = df['score'].quantile(0.75)
IQR = Q3 - Q1

lower = Q1 - 1.5 * IQR
upper = Q3 + 1.5 * IQR

outliers = df[(df['score'] < lower) | (df['score'] > upper)]
print('Lower fence:', lower)
print('Upper fence:', upper)
print('Number of outliers:', len(outliers))

Lower fence: 3.7500000000000004
Upper fence: 8.95
Number of outliers: 51


By using the IQR method we find that the lower fence is at 3.75 whereas the upper fence is 8.95 and there are 51 outliers meaning there are 51 movies that are out of this range. This shows that there are not errors but extreme cases of good and bad movies.

Wrong Formatts

In [13]:
print(df.dtypes)

name             str
rating           str
genre            str
year           int64
released         str
score        float64
votes          int64
director         str
writer           str
star             str
country          str
budget         int64
gross        float64
company          str
runtime,,        str
dtype: object


Two format issues were found. The runtime column has an error in the name (runtime,,) and values with trailing commas. The released column stores dates as plain text instead of a correct and proper date format.

**Basic Statistics**

In [14]:
df.describe()

,year,score,votes,budget,gross
count,4000.000000,4000.000000,4.000000e+03,4.000000e+03,3.831000e+03
mean,1991.191500,6.321950,5.449450e+04,1.542740e+07,4.172202e+07
std,5.934529,0.977101,1.427738e+05,2.248639e+07,9.259504e+07
min,1980.000000,2.200000,5.100000e+01,0.000000e+00,3.090000e+02
25%,1986.000000,5.700000,4.300000e+03,0.000000e+00,2.835743e+06
50%,1991.000000,6.400000,1.300000e+04,7.000000e+06,1.183822e+07
75%,1996.000000,7.000000,4.500000e+04,2.100000e+07,3.560495e+07
max,2001.000000,9.300000,2.400000e+06,2.000000e+08,2.201647e+09


In [15]:
print('Mean:  ', df['score'].mean())
print('Median:', df['score'].median())
print('Min:   ', df['score'].min())
print('Max:   ', df['score'].max())

Mean:   6.321950000000001
Median: 6.4
Min:    2.2
Max:    9.3


this shows us our mean, median, min and max

In [16]:
print(df['genre'].value_counts())

genre
Comedy       1355
Drama         807
Action        775
Crime         302
Adventure     231
Biography     175
Horror        158
Animation     113
Fantasy        33
Mystery        13
Thriller       11
Family          8
Romance         8
Sci-Fi          6
Western         3
Music           1
History         1
Name: count, dtype: int64
